cfme/tests/infrastructure/test_vm_power_control.py

```diff
@@ -403,9 +403,15 @@ def vm(self, rest_api, vm_name):
         assert result.name == vm_name
         return result
 
-    def test_power_off(self, verify_vm_running, vm):
+    @pytest.mark.parametrize(
+        "from_detail", [True, False],
+        ids=["delete_from_detail", "delete_from_collection"])
+    def test_stop(self, rest_api, verify_vm_running, vm, from_detail):
         assert "stop" in vm.action
-        vm.action.stop()
+        if from_detail:
+            vm.action.stop()
+        else:
+            rest_api.collections.vms.action.stop(vm)
         wait_for(lambda: vm.power_state == "off", num_sec=300, delay=5, fail_func=vm.reload)
```

Comment:
Hmm could we add a message to ease the failure troubleshooting here?
Something like `message='wait for VM to stop (current state: {})'.format(vm.power_state)`
And in the rest of the power control functions as well, just in case.